File: src/my_py_pkg/my_py_pkg/smartphone.py
#!/usr/bin/env python3
# flake8: noqa
"""
This is a simple node that publishes a string message to the topic "my_first_topic"
"""


import rclpy
from rclpy.node import Node
from example_interfaces.msg import String


class SmartphoneNode(Node):
    def __init__(self):
        super().__init__("smartphone")

        self.subscriber_ = self.create_subscription(
            String,
            "/robot_news",
            self.callback_robot_news,
            10,
        )
        self.get_logger().info("Smartphone has been started")

    def callback_robot_news(self, msg):
        self.get_logger().info(f"I received: {msg.data}")


def main(args=None):
    rclpy.init(args=args)
    node = SmartphoneNode()
    rclpy.spin(node)  # mantém o programa rodando
    rclpy.shutdown()


if __name__ == "__main__":
    main()

File: src/my_cpp_pkg/src/smartphone.cpp
#include "example_interfaces/msg/string.hpp"
#include "rclcpp/rclcpp.hpp"

class RobotNewsStationNode : public rclcpp::Node {
public:
  RobotNewsStationNode() : Node("smartphone") {
    subscriber_ = this->create_subscription<example_interfaces::msg::String>(
        "robot_news", 10,
        std::bind(&RobotNewsStationNode::callbackRobotNews, this,
                  std::placeholders::_1));

    RCLCPP_INFO(this->get_logger(), "Smartphone has been started");
  }

private:
  void callbackRobotNews(const example_interfaces::msg::String::SharedPtr msg) {
    RCLCPP_INFO(this->get_logger(), "%s", msg->data.c_str());
  }

  rclcpp::Subscription<example_interfaces::msg::String>::SharedPtr subscriber_;
};

int main(int argc, char const *argv[]) {
  rclcpp::init(argc, argv);
  auto node = std::make_shared<RobotNewsStationNode>();
  rclcpp::spin(node);
  rclcpp::shutdown();
  return 0;
}
